src/ui/preferences/UdpSettings.qml

```diff
@@ -51,27 +66,23 @@ Column {
     }
     Item {
         height: ScreenTools.defaultFontPixelHeight / 2
-        width:  parent.width
+        width:  ScreenTools.defaultFontPixelWidth * 43
     }
     QGCLabel {
-        text:   qsTr("Target Hosts:")
+        text:   qsTr("Server addresses (optional):")
     }
     Item {
         width:  hostRow.width
         height: hostRow.height
         Row {
             id:      hostRow
             spacing: ScreenTools.defaultFontPixelWidth
-            Item {
-                height: 1
-                width:  _firstColumn
-            }
             Column {
                 id:         hostColumn
                 spacing:    ScreenTools.defaultFontPixelHeight / 2
                 Rectangle {
                     height:  1
-                    width:   _secondColumn
+                    width:   ScreenTools.defaultFontPixelWidth * 43
```

Comment:
Why rectangle is using font pixel size to to derive its size ?
It should use anchors, or someone else size.

Reply:
You are correct, and I made the same mistake in several other places. I will do a commit soon replacing multiple width definitions with `parent.width` instead. However there is one place where I can't do this and that is on row 134. I tried to use both `parent.width` and `hostRow.width` but it messes up the buttons' layout. The only thing I got working correctly was using either `ScreenTools.defaultFontPixelWidth * 43` or `_firstColumn + _secondColumn`. Is the second way of doing it considered "correct"? Or do you know a better way of doing it?
Thanks!

Reply:
Wow this qml is ancient history with respect to crazy nastiness in layout. I think adding together the first and second column is the only sane way to go here beyond rewriting the whole thing.

Reply:
Haha I see. Well I will start there then and could maybe look into a potential rewrite when I've got myself a few more hours into this project. A commit is arriving soon.